src/ray/raylet/node_manager.cc

```diff
@@ -1251,11 +1241,7 @@ void NodeManager::EnqueuePlaceableTask(const Task &task) {
   // (See design_docs/task_states.rst for the state transition diagram.)
   if (args_ready) {
     local_queues_.QueueReadyTasks({task});
-    // Dispatch just the new who was added to the ready task.
-    // The other tasks in the ready queue can be ignored as no new resources
-    // have been added and no new worker has became available since the last
-    // time DispatchTasks() was called.
-    DispatchTasks({task});
+    DispatchTasks();
```

Comment:
Hmm is there a way to avoid iterating over the entire `READY` queue here? I liked that the code was explicit about which tasks might be able to be scheduled before. One way to keep the code common would be to have an `DispatchTasks()` helper method that takes in a list of task IDs, and the `ResourceSet` that all tasks in the list require.

Reply:
This is not iterating through the entire READY queue, that's the nice thing about this PR. It just looks at the resource shapes and only looks at as many tasks as can be scheduled at once.